exp/api/v1beta2/awsmanagedmachinepool_types.go

```diff
@@ -297,6 +301,15 @@ type AWSManagedMachinePoolStatus struct {
 	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
 }
 
+// NodeRepairConfig defines the node auto repair configuration for managed node groups.
+type NodeRepairConfig struct {
```

Comment:
Is there a chance to extend this struct (adding more fields) ? if not, then adding autoNodeRepair boolean field under spec is better

Reply:
There is are more fields, see https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/eks@v1.74.2/types#NodeRepairConfig Do you want me to add support for all?

Reply:
Fine as a struct, given that more options may be supported in the future. Is `enabled` enough as default to get the feature to work nicely? If yes, I think we can live with only the `enabled` field for now.

Reply:
Yes, those field did not exist in 1.73.0 version/until recently. This is why i did not include them from the start see https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/eks@v1.73.0/types#NodeRepairConfig

Also the API linked above does not mention any being required, see the `Required: No`